### PR TITLE
Upgraded OGRE version to v1.12.12

### PR DESCRIPTION
**Pull request by varunagrawal**
Taking over from #700.

### CODE REVIEW

File: rviz_ogre_vendor/package.xml [hunk 1]
```diff
@@ -30,6 +30,7 @@
   <depend>libxaw</depend>
   <depend>libxrandr</depend>
   <depend>opengl</depend>
+  <depend>pugixml-dev</depend>
```
Comment (clalancette) [inline rviz_ogre_vendor/package.xml:33]: Is this a new dependency of OGRE 1.12.12?  Can we avoid this dependency?  If not, we'll have to figure out a way to package and/or vendor it for Windows and macOS.

Comment (sundermann) [reply rviz_ogre_vendor/package.xml:33]: It is not. It is build with Ogre: https://github.com/OGRECave/ogre/blob/v1.12.12/CMake/Dependencies.cmake#L143

**Comment by sundermann** [comment]
What's the status on this PR?

**Comment by clalancette** [comment]
> What's the status on this PR?

In short, all of the warnings that this is going to produce need to be fixed to get this in.  I did some work towards this in https://github.com/ros2/rviz/tree/clalancette/ogre-1.12.10-upgrade , but it is completely untested (I'm not sure if it even compiles at the moment).  Someone needs to go through that set of changes, make it compile, test it out, and then submit an update to this PR (or a new PR) which does the update.

**Comment by varunagrawal** [comment]
I unfortunately don't have the cycles for this and it is more challenging for me than I had originally hoped. :(

**Pull request closed without merging**